src/media/local-stream.ts

```diff
@@ -60,6 +79,50 @@ abstract class _LocalStream extends Stream {
       this[StreamEventNames.Muted].emit(isMuted);
     }
   }
+
+  /**
+   * Adds an effect to a local stream.
+   *
+   * @param name - The name of the effect.
+   * @param effect - The effect to add.
+   */
+  async addEffect(name: string, effect: TrackEffect): Promise<void> {
+    const outputTrack = await effect.load(this.outputStream.getTracks()[0]);
+    this.effects.push({ name, effect });
+    replaceTrack(this.outputStream, outputTrack);
+
+    // When the effect's track is updated, update the next effect or output stream.
+    effect.on(EffectEvent.TrackUpdated, (track: MediaStreamTrack) => {
+      const effectIndex = this.effects.findIndex((i) => i.name === name);
```

Comment:
nit: use `e` instead of `i` here? `i` makes me think it's an index.

Reply:
Updated.